### ISSUE TITLE
NC11 + LDAP, Loss of performance

### ISSUE THREAD

**Issue by ghost** [nextcloud/server#3463]
Nextcloud version (eg, 10.0.2): 11.0.1
Operating system and version (eg, Ubuntu 16.04): CentOS 7
Apache or nginx version (eg, Apache 2.4.25): Apache 2.4.6
PHP version (eg, 5.6): php5.6
Database: PgSQL 9.4
Users: LDAP

Replication:
Every time I do upgrade from NC 10 to 11 I'm facing a problem with the process. The upgrade procedure always exits with Segmentantion Fault error which points to LDAP module. We found a way to bypass this by changing two variables in user_ldap/lib/Access.php: 
$maxResults = ... 500 to 0
$pageSize = ... 500 to 0

Then we can proceed with upgrade.

The issue you are facing: 
After moving to NC11 the whole system slows down drastically (as opposed to official statement). When logging in, trying to view users, run Administration panel, everything goes slow, and looking at processes, database seems to work full throttle. I have backup of v10.0.1 and both DBs have around 700 users(active), in LDAP around 8k. On NC10 everything runs smoothly. There was no modification made in config.

Config below:

```
<?php
$CONFIG = array (
'instanceid' => *
'passwordsalt' => *
'secret' => *',
'trusted_proxies' =>
array (
0 => '*',
),
'trusted_domains' =>
array (
0 => '*',
1 => 'localhost',
2 => *
),
'datadirectory' => '/var/nextcloud-data',
'overwrite.cli.url' => '*',
'htaccess.RewriteBase' => '*',
'dbtype' => 'pgsql',
'version' => '11.0.1.2',
'dbname' => '*',
'dbhost' => 'localhost',
'dbport' => '',
'dbtableprefix' => 'oc_',
'dbuser' => *
'dbpassword' => *
'memcache.local' => '\OC\Memcache\APCu',
'logtimezone' => 'UTC',
'default_language' => 'pl',
'lost_password_link' => '*',
'installed' => true,
'ldapIgnoreNamingRules' => false,
'ldapProviderFactory' => '\OCA\User_LDAP\LDAPProviderFactory',
'appstore.experimental.enabled' => true,
'mail_smtpmode' => 'smtp',
'mail_from_address' => '*',
'mail_domain' => '*',
'mail_smtphost' => '*',
'mail_smtpport' => '25',
'loglevel' => 2,
'maintenance' => false,
'integrity.check.disabled' => true,
);
```

**Comment by LukasReschke** [comment]
cc @blizzz

**Comment by ryannix123** [comment]
I'm seeing this as well, albeit with the following configuration:

Nextcloud version 11.0.1
Operating system and version (eg, Ubuntu 16.04): CentOS 7
Apache or nginx version (eg, Apache 2.4.25): Apache 2.4.6
PHP version (eg, 5.6): php7.0
Database: MariaDB 10.1
Users: Active Directory

**Comment by blizzz** [comment]
> If you have LDAP users in your NC instance, and if by any chance their DN changes, the user_ldap app goes into infinite loop, all the time looking for the original DN. Memory which the PHP script consumes, at some point reaches the maximum defined in your php.ini config and it exits with SIGSEGV (Segmentation Fault).

Unsure whether this is totally related, but I found a regression and fixed it in 11.0.2. It requires however to re-check the UUID of each mapped LDAP user. This is done via background jobs and could take a bit, depending on the amount of users. See https://github.com/nextcloud/server/pull/3528 (details in the description of #3521)

**Comment by ChristophWurst** [comment]
ghost -> close
